src/util/hashids.rs

```diff
@@ -16,3 +20,15 @@ pub fn to_u64(hash_id: &str) -> Result<u64, &str> {
     let id = ids.first().ok_or("No ID found in hash ID")?;
     Ok(*id)
 }
+
+pub fn alias_comparator(id: &str) -> Box<dyn Fn(&Pasta) -> bool> {
```

Comment:
why does this return a function? that seems overly complicated. Further more the name is a bit generic. To me its not clear at all what the function does given its name. Half of the logic living in the returned closure does not help parsing it.

I might be missing something but why is the returned closure used everywhere else as "comparator"?

Reply:
This function is to find the entry that is equal to the hash_id, animal_id or custom_url in the link. Because there are many places that use the same and repeated comparison logic, I have separated it out. I am not good at naming, maybe it should be renamed